UpdateMavenModel does not handle valid XML properly

## What version of OpenRewrite are you using?

I am using

- OpenRewrite v8.11.1
- rewrite-maven_ v8.11.1

## How are you running OpenRewrite?

I am using the Maven plugin, and my project is a multi-module project. However, the issue occurs also with single module projects and is demonstrated in a simple unit test below.

For reasons outside my control I have Maven `pom.xml` files that use XML-escaped characters in Maven coordinates. This leads to the stacktrace documented below ("Dependency must have artifactId").

## What is the smallest, simplest way to reproduce the problem?

```java
package demo;

import static org.openrewrite.maven.Assertions.pomXml;

import org.junit.jupiter.api.Test;
import org.openrewrite.maven.AddManagedDependency;
import org.openrewrite.test.RecipeSpec;
import org.openrewrite.test.RewriteTest;

class ExampleTest implements RewriteTest {

  @Override
  public void defaults(final RecipeSpec spec) {
    spec.recipe(new AddManagedDependency("ch.qos.logback", "logback-classic", "1.4.14", null, null, null, null, null, null, null));
  }

  @Test
  void shouldNotFailWithValidMavenXml() {
    rewriteRun(pomXml("""
      <project>
        <groupId>com.mycompany.app</groupId>
        <artifactId>my-app</artifactId>
        <version>1</version>
        <dependencies>
          <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-&#0097;pi</artifactId>
            <version>2.0.9</version>
          </dependency>
        </dependencies>
      </project>
      """,
      """
      <project>
        <groupId>com.mycompany.app</groupId>
        <artifactId>my-app</artifactId>
        <version>1</version>
        <dependencyManagement>
          <dependencies>
            <dependency>
              <groupId>ch.qos.logback</groupId>
              <artifactId>logback-classic</artifactId>
              <version>1.4.14</version>
            </dependency>
          </dependencies>
        </dependencyManagement>
        <dependencies>
          <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-&#0097;pi</artifactId>
            <version>2.0.9</version>
          </dependency>
        </dependencies>
      </project>
      """
      ));
  }

}
```

## What did you expect to see?

A successful modification.

## What did you see instead? What is the full stack trace of any errors you encountered?

```
Caused by: java.lang.IllegalStateException: Dependency must have artifactId
  at org.openrewrite.maven.UpdateMavenModel.lambda$visitDocument$1(UpdateMavenModel.java:70)
  at java.base/java.util.Optional.orElseThrow(Optional.java:403)
  at org.openrewrite.maven.UpdateMavenModel.visitDocument(UpdateMavenModel.java:70)
  at org.openrewrite.xml.tree.Xml$Document.acceptXml(Xml.java:151)
  at org.openrewrite.xml.tree.Xml.accept(Xml.java:50)
  at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:278)
  ... 88 more
```

The reason for this is the (naive?) use of `Xml.Tag.getChildValue()` which at its core determines the expected text content [here](https://github.com/openrewrite/rewrite/blob/v8.11.1/rewrite-xml/src/main/java/org/openrewrite/xml/tree/Xml.java#L357-L368); I did not check, but there seem to many such places, for example [here](https://github.com/openrewrite/rewrite/blob/v8.11.1/rewrite-maven/src/main/java/org/openrewrite/maven/UpdateMavenModel.java#L54).

The problem with XML-escaped characters is now that this is no longer a list of size one, but a list of text fragments.
For example for `<artifactId>slf4j-&#0097;pi</artifactId>` this is a list of `slf4j-`, `&#0097;` and `pi` - still valid XML, but it won't be properly processed.

I am unsure what a proper "fix" might be. Should the `Xml.Tag` have an additional method `getInnerText()` (similar as in HTML DOM elements)? Should `UpdateMavenModel` get the strings for Maven coordinates in a more robust manner itself with some helper method?

## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?

I'd rather not, as I have trouble getting everything to compile on my private setup. Additionally, I am too unfamiliar with the code to make the "right" changes, I expect.

--- a/rewrite-maven/src/test/java/org/openrewrite/maven/MavenParserTest.java
+++ b/rewrite-maven/src/test/java/org/openrewrite/maven/MavenParserTest.java
@@ -29,7 +29,6 @@ import org.openrewrite.maven.internal.MavenParsingException;
 import org.openrewrite.maven.tree.*;
 import org.openrewrite.test.RewriteTest;
 import org.openrewrite.tree.ParseError;
-import org.openrewrite.xml.tree.Xml;
 
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
@@ -2298,5 +2297,53 @@ class MavenParserTest implements RewriteTest {
         assertThat(MavenParser.builder().build().parse(malformedPomXml))
           .singleElement()
           .isInstanceOf(ParseError.class);
+    }
+
+    @Test
+    @Issue("https://github.com/openrewrite/rewrite/issues/3811")
+    void escapedA() {
+        rewriteRun(
+          spec -> spec.recipe(new AddManagedDependency("ch.qos.logback", "logback-classic", "1.4.14", null, null, null, null, null, null, null)),
+          //language=xml
+          pomXml(
+            """
+              <project>
+                <groupId>com.mycompany.app</groupId>
+                <artifactId>my-app</artifactId>
+                <version>1</version>
+                <dependencies>
+                  <dependency>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>slf4j-&#0097;pi</artifactId>
+                    <version>2.0.9</version>
+                  </dependency>
+                </dependencies>
+              </project>
+              """,
+            """
+              <project>
+                <groupId>com.mycompany.app</groupId>
+                <artifactId>my-app</artifactId>
+                <version>1</version>
+                <dependencyManagement>
+                  <dependencies>
+                    <dependency>
+                      <groupId>ch.qos.logback</groupId>
+                      <artifactId>logback-classic</artifactId>
+                      <version>1.4.14</version>
+                    </dependency>
+                  </dependencies>
+                </dependencyManagement>
+                <dependencies>
+                  <dependency>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>slf4j-&#0097;pi</artifactId>
+                    <version>2.0.9</version>
+                  </dependency>
+                </dependencies>
+              </project>
+              """
+          )
+        );
     }
 }

--- a/rewrite-xml/build.gradle.kts
+++ b/rewrite-xml/build.gradle.kts
@@ -23,6 +23,7 @@ dependencies {
 
     implementation("org.antlr:antlr4:4.11.1")
     implementation("io.micrometer:micrometer-core:1.9.+")
+    implementation("org.apache.commons:commons-text:1.11.+")
 
     testImplementation(project(":rewrite-test"))
     testImplementation(project(":rewrite-maven"))

--- a/rewrite-xml/src/main/java/org/openrewrite/xml/tree/Xml.java
+++ b/rewrite-xml/src/main/java/org/openrewrite/xml/tree/Xml.java
@@ -17,6 +17,7 @@ package org.openrewrite.xml.tree;
 
 import lombok.*;
 import lombok.experimental.FieldDefaults;
+import org.apache.commons.text.StringEscapeUtils;
 import org.intellij.lang.annotations.Language;
 import org.openrewrite.*;
 import org.openrewrite.internal.lang.Nullable;
@@ -358,11 +359,14 @@ public interface Xml extends Tree {
             if (content == null) {
                 return Optional.empty();
             }
-            if (content.size() != 1) {
-                return Optional.empty();
-            }
-            if (content.get(0) instanceof Xml.CharData) {
+            if (content.size() == 1 && content.get(0) instanceof Xml.CharData) {
                 return Optional.ofNullable(((CharData) content.get(0)).getText());
+            }
+            if (content.stream().allMatch(c -> c instanceof Xml.CharData)) {
+                return Optional.of(content.stream()
+                        .map(c -> ((CharData) c).getText())
+                        .map(StringEscapeUtils::unescapeXml)
+                        .collect(Collectors.joining()));
             }
             return Optional.empty();
         }

